lib/views/question/question_view.dart

```diff
@@ -0,0 +1,115 @@
+
+  const QuestionView({
+    super.key,
+    required this.uiModel,
+    required this.child,
+    required this.onNextQuestion,
+    required this.onSubmit,
+  });
+
+  AppBar _appBar(BuildContext context) => AppBarExt.appBarWithCloseButton(
+        context: context,
+        onPressed: () => context.goNamed(
+          RoutePath.home.name,
+        ),
+      );
+
+  Widget _background(BuildContext context) => SizedBox(
+        width: MediaQuery.of(context).size.width,
+        height: MediaQuery.of(context).size.height,
+        // TODO: - Bind question's photo
+        child: Image(image: Assets.images.nimbleBackground.image().image),
+      );
+
+  Widget _mainBody(BuildContext context) => Container(
+        width: MediaQuery.of(context).size.width,
+        padding: const EdgeInsets.all(20),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.start,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            _questionNumber(context),
+            const SizedBox(
+              height: 16,
+            ),
+            _questionTitle(context),
+            SingleChildScrollView(child: child),
+          ],
+        ),
+      );
+
+  Widget _questionNumber(BuildContext context) =>
+      Consumer(builder: (_, ref, __) {
+        final state = ref.watch(questionViewModelProvider);
+        return state.maybeWhen(
+          orElse: () => const Text(''),
+          success: (uiModel) => Text(
+            '${uiModel.currentQuestionIndex}/${uiModel.totalQuestions}',
+            style: Theme.of(context).textTheme.bodyMedium,
+          ),
+        );
+      });
+
+  Widget _questionTitle(BuildContext context) => Text(
+        uiModel.title,
+        style: Theme.of(context).textTheme.displayMedium,
+      );
+
+  Widget get _floatingActionButton {
+    final isLastQuestion = uiModel.totalQuestions > 0 &&
+        uiModel.currentQuestionIndex == uiModel.totalQuestions;
+    return isLastQuestion ? _submitButton : _nextButton;
+  }
+
+  Widget get _nextButton => FloatingActionButton(
+        foregroundColor: Colors.black,
+        backgroundColor: Colors.white,
+        child: const Icon(Icons.navigate_next),
+        onPressed: () => _nextQuestion(),
+      );
+
+  Widget get _submitButton => ElevatedButton(
+        style: ElevatedButton.styleFrom(minimumSize: const Size(140, 56)),
+        onPressed: () => _submit(),
+        child: const Text('Submit'),
+      );
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      children: [
+        _background(context),
+        Container(color: Colors.black38),
+        Scaffold(
+          backgroundColor: Colors.transparent,
+          appBar: _appBar(context),
+          body: SafeArea(child: _mainBody(context)),
+          floatingActionButton: _floatingActionButton,
+          floatingActionButtonLocation: FloatingActionButtonLocation.endDocked,
+        )
+      ],
+    );
+  }
+
+  void _nextQuestion() {
```

Comment:
@nmint8m I think we can use `onNextQuestion` directly, no need to define more functions, it is cumbersome 🙏